packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -205,6 +206,65 @@ export class BlobManager {
         });
     }
 
+    /**
+     * Generates data used for garbage collection. Each blob uploaded represents a node in the GC graph as it can be
+     * individually referenced by storing its handle in a referenced DDS. Returns the list of blob ids as GC nodes.
+     * @param fullGC - true to bypass optimizations and force full generation of GC data. BlobManager doesn't care
+     * about this for now because the data is a simple list of blob ids.
+     */
+    public getGCData(fullGC: boolean = false): IGarbageCollectionData {
+        const getGCNodePath = (blobId: string) => { return `/${BlobManager.basePath}/${blobId}`; };
+        const gcData: IGarbageCollectionData = { gcNodes: {} };
+        /**
+          * The node path is of the format `/_blobs/blobId`. This path must match the path of the blob handle returned
+          * by the createBlob API because blobs are marked referenced by storing these handles in a referenced DDS.
+          */
+        this.blobIds.forEach((blobId: string) => {
+            gcData.gcNodes[getGCNodePath(blobId)] = [];
+        });
+
+        /**
+         * For all blobs in the redirect table, the handle returned on creation is based off of the localId. So, these
+         * nodes can be referenced by storing the localId handle. When that happens, the corresponding storageId node
+         * must also be marked referenced. So, we add a route from the localId node to the storageId node.
+         * Note that because of de-duping, there can be multiple localIds that all redirect to the same storageId or
+         * a blob may be referenced via its storageId handle.
+         */
+        if (this.redirectTable !== undefined) {
+            for (const [localId, storageId] of this.redirectTable) {
+                // Add node for the localId and add a route to the storageId node. The storageId node will have been
+                // added above when adding nodes for this.blobIds.
+                gcData.gcNodes[getGCNodePath(localId)] = [getGCNodePath(storageId)];
```

Comment:
Will this ever overwrite anything we don't want overwritten?

Reply:
This code is adding node for each key in the redirect table which should be unique. The only problem can be if a localId is the same as storageId. This should never happen and even if it did, it should be fine because the node would end up having a route to itself.
If the same localId could redirect to multiple storageIds, then we cannot do this and have to append paths. But that's not possible so this should be fine.

Reply:
Should we then have some sort of validation here? Maybe some telemetry or asserts?

Reply:
I don't think so. `this.redirectTable` is a 1:1 map of localId -> storageId. So, we cannot have one localId redirecting to multiple storageIds.